docs/internal/investigations/secure-resolvers/secure-resolvers-v2.md

```diff
@@ -1,39 +1,41 @@
 
-There are certain integration systems (such as UI applications) that require unrestricted access to the Director API. They should keep working properly having a global view/access.
+Some integration systems (such as UI applications) require unrestricted access to the Director API. They should keep working properly having a global view/access.
 
-There is an implicit requirement derived from the requirements above that no system should be able to list all applications, runtimes or integration systems.
+An implicit requirement, derived from the requirements above, is that no system must be allowed to list all applications, runtimes, or integration systems.
 
-(Out-of-scope) Limit access to the `auths` field of each type to admin Users only.
+Out-of-scope: Limit access to the `auths` field of each type to admin users only.
 
 ## Current State
 
-The following section will analyze the problem in all current queries and mutations and group them in sections providing a conceptual solution.
+The following section analyzes the problem in all current queries and mutations, and groups them in sections providing a conceptual solution.
 
-The problem with securing resolvers is three dimencional (mutations/queries X consumer X owner) There are about 40 mutations/queries that have to be secured, 4 consumer types and 4 owner types. For specific queries/mutations there is always a single owner but what is tricky is how to derive the id of that owner. For example, if in getApplication(applicationID), the owner id is passed as input to the query while in updatePackage(packageID) some special code needs to be executed to find the application id from the package id. In the case of updateAPIDefinition(apiDefID) the special code that needs to be executed to derive.
+The problem with securing resolvers is three dimensional. It features all three: mutations/queries; consumer, and owner. In total, there are about 40 mutations/queries that have to be secured, 4 consumer types, and 4 owner types. For specific queries/mutations there is always a single owner but obtaining the id of that owner is not always straight forward. For example, in getApplication(applicationID) the owner id is passed as input to the query, however, in updatePackage(packageID) some special code needs to be executed to find the application id from the package id. In the case of updateAPIDefinition(apiDefID) the special code that needs to be executed to derive (nck: To derive what?).
```

Comment:
In the case of updateAPIDefinition(apiDefID) the special code that needs to be executed to derive (nck: To derive what?). -> just remove it, the prev example is enough. You can put `updatePackage(packageID) or  updateAPIDefinition(apiDefID)` in the previous sentence